pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -304,6 +390,8 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 	status.CollisionCount = new(int32)
 	*status.CollisionCount = collisionCount
 
+	updateStatus(&status, set.Spec.MinReadySeconds, currentRevision, updateRevision, pods)
```

Comment:
Nit: We may consider using named parameters and writing here
```
defer updateStatus(&status, set.Spec.MinReadySeconds, currentRevision, updateRevision, pods)
```

this way, any place we leave the function the update will be always updated.

Reply:
I thought about this before, but this would also modify status after calling updateStatefulSetAfterInvariantEstablished(), and we don't want to do that.

BTW I think this would work even without named parameters: https://go.dev/play/p/VIqxr6zW6u2